addons/purchase_stock/models/stock.py

```diff
@@ -67,7 +67,7 @@ def _generate_valuation_lines_data(self, partner_id, qty, debit_value, credit_va
 
         rslt = super(StockMove, self)._generate_valuation_lines_data(partner_id, qty, debit_value, credit_value, debit_account_id, credit_account_id, description)
         if self.purchase_line_id:
-            purchase_currency = self.purchase_line_id.currency_id
+            purchase_currency = self.purchase_line_id.order_id.currency_id
```

Comment:
why do you need this ?

Reply:
currency_id is related field to order.currency 
but for some reason it was not set during the test

Reply:
I think it might be because of the flush